tests/integration/utils/flow_tools.py

```diff
@@ -74,7 +82,10 @@ def _make_flow(
         .setdefault('default run length', 'PT0S')
     )
     # allow implicit tasks by default:
-    conf.setdefault('scheduler', {}).setdefault('allow implicit tasks', 'True')
+    if defaults:
```

Comment:
Should add the other default set above into this block.

Reply:
I have changed it for completeness, I don't think that's necessary, because `[runtime][NAMESPACE][simulation]default run length is both unchanged from Cylc 7 and overrideable.

Reply:
(It's not relevant to your use case, but will likely be relevant to other validate/config use cases)